public/src/components/amounts/AmountsForm.tsx

```diff
@@ -0,0 +1,174 @@
+} from '../../utils/requests';
+import withS3Data, { InnerProps, DataFromServer } from '../../hocs/withS3Data';
+
+import { makeStyles, Theme } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
+  body: {
+    display: 'flex',
+    overflow: 'hidden',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%',
+  },
+  leftCol: {
+    height: '100%',
+    flexShrink: 0,
+    overflowY: 'auto',
+    background: 'white',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+    paddingRight: spacing(6),
+  },
+  rightCol: {
+    overflowY: 'auto',
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    paddingTop: spacing(6),
+    paddingLeft: spacing(6),
+  },
+}));
+
+const AmountsForm: React.FC<InnerProps<AmountsTests>> = ({
+  data: configuredAmounts,
+  setData: setConfiguredAmounts,
+  saveData: saveConfiguredAmounts,
+}: InnerProps<AmountsTests>) => {
+  const classes = useStyles();
+
+  const [selectedTest, setSelectedTest] = useState<AmountsTest | undefined>();
+
+  const onTargetSelected = (target: Territory) => {
+    const currentTest = configuredAmounts.find(test => test.target === target);
+    if (currentTest) {
+      setSelectedTest(currentTest);
+    } else {
+      setSelectedTest(undefined);
+    }
+  };
+
+  const createLocalTest = (selected: CountryOptions) => {
+    if (selected && selected.label && selected.code) {
+      const newTest = {
+        testName: `SUPPORTER_AMOUNTS_EVERGREEN__${selected.code}`,
+        liveTestName: `SUPPORTER_AMOUNTS_AB_TEST__${selected.code}`,
+        isLive: false,
+        target: selected.code,
+        // Need to calculate seed
+        seed: Math.floor(Math.random() * 1000000),
+        variants: [
+          {
+            variantName: 'CONTROL',
+            defaultContributionType: 'MONTHLY' as ContributionType,
+            displayContributionType: [
+              'ONE_OFF' as ContributionType,
+              'MONTHLY' as ContributionType,
+              'ANNUAL' as ContributionType,
+            ],
+            amountsCardData: {
+              ONE_OFF: {
+                amounts: [1],
+                defaultAmount: 1,
+                hideChooseYourAmount: false,
+              },
+              MONTHLY: {
+                amounts: [10],
+                defaultAmount: 10,
+                hideChooseYourAmount: false,
+              },
+              ANNUAL: {
+                amounts: [100],
+                defaultAmount: 100,
+                hideChooseYourAmount: false,
+              },
+            },
+          },
+        ],
+      };
+
+      const updatedTests = [...configuredAmounts, newTest];
+      setConfiguredAmounts(updatedTests);
+      setSelectedTest(newTest);
+    }
+  };
+
+  const updateLocalTest = (updated: AmountsTest) => {
+    const updatedTests = configuredAmounts.filter(t => t.testName !== updated.testName);
+    updatedTests.push(updated);
```

Comment:
Does the ordering here matter at all? Is it okay that the updated test always moves to the end of the array?

Reply:
No, ordering is not an issue here. It's just an array of tests

Reply:
👍  As mentioned elsewhere, if order isn't important then perhaps an object indexed by some unique identifier (`testName`?) would be a better choice over an array? It'd make lookups more straightforward too. Perhaps one for a future PR if that's out of scope here.